bcgov/PIMS
# PIMS-1750: Modify ETL to extract metadata values into appropriate tables/columns

## 🎯 Summary

  
[PIMS-1750: ](https://citz-imb.atlassian.net/browse/PIMS-1750)


**ETL Changes:**
No longer importing the "LeasedLandMetadata" field from the Buildings table or the "Metadata" column from the ProjectSnapshots table.

Added the OCG gain loss metadata field to the MonetaryType table in Postgres, and changed the CompletedOn date column in ProjectTasks to a regular column to because a date was being created even when the “isCompleted” boolean value was false.

Created a data migration to handle the alter project_task table, and modified the “1717629146741-MoveMetadataToTables.ts” migration to include the OCGGainLoss field as a monetary type.


**Steps for Testing:**
To test this locally, you'll need to run these commands to drop and recreate your local Postgres schema:

`DROP SCHEMA PUBLIC CASCADE`
`CREATE SCHEMA PUBLIC`

Then stop and restart the express api backend.

Then copy the tasks.json in this PR to replace your existing tasks.json in this directory: `"C:\Users\lawrlau\AppData\Roaming\DBeaverData\workspace6\General\.dbeaver"`
For Mac users, I believe it’s this: `“~/Library/DBeaverData/"`

Once you have copied the new tasks.json file, just start your local DBeaver application, navigate to your "Database Tasks" section, or go to Windows->Database Tasks, if it isn't visible in DBeaver.
Then, right click on the Prod to Local task and click on "Run Task":
![image](https://github.com/bcgov/PIMS/assets/68400651/71803756-3213-4cca-90e0-283047732ff9)

This should be tested thoroughly, to make sure we've got all the metadata values associated to the correct data types, etc, as I did a lot of copying and pasting and might have missed something or added something incorrectly.

Note: after running the ETL, you may need to run the username updater script if you find yourself stuck at the access request page.

`cd ../tools/usernameUpdater/`

`npm run start`

## 🔰 Checklist


> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.
